integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py

```diff
@@ -0,0 +1,194 @@
+        validate_response = EthGetBlockByHashResult(**response)
+
+        assert hex(contract.functions.getBlockTimestamp().call()) == validate_response.result.timestamp
+
+    def test_block_timestamp_event(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[1]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.call_event().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        event_logs = contract.events.Result().process_receipt(resp)
+        assert len(event_logs) == 1
+        assert hex(event_logs[0]["args"]["block_timestamp"]) == validate_response.result.timestamp
+
+    def test_block_timestamp_iterative(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[1]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.call_events_in_loop().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        event_logs = contract.events.Result().process_receipt(resp)
+        assert len(event_logs) == 1
+        assert hex(event_logs[0]["args"]["block_timestamp"]) == validate_response.result.timestamp
+
+    def test_block_timestamp_constructor(self, block_timestamp_contracts, json_rpc_client):
+        contract, resp = block_timestamp_contracts[2]
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        assert hex(contract.functions.getBlockTimestamp().call()) == validate_response.result.timestamp
+
+    def test_block_timestamp_mapping_event(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[3]
+        sender_account = self.accounts[0]
+
+        text = "".join([random.choice(string.ascii_uppercase) for _ in range(5)])
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.addData(text, 1).build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        event_logs = contract.events.DataAdded().process_receipt(resp)
```

Comment:
I think we don't need this test 
test_block_timestamp_mapping_get_data - almost the same but better